sys/syz-sysgen/sysgen.go

```diff
@@ -274,20 +274,48 @@ func generateExecutorSyscalls(target *targets.Target, syscalls []*prog.Syscall,
 				last = i
 			}
 		}
-		data.Calls = append(data.Calls, SyscallData{
-			Name:     c.Name,
-			CallName: c.CallName,
-			NR:       int32(c.NR),
-			NeedCall: (!target.SyscallNumbers || strings.HasPrefix(c.CallName, "syz_")) && !c.Attrs.Disabled,
-			Attrs:    attrVals[:last+1],
-		})
+		data.Calls = append(data.Calls, newSyscallData(target, c, attrVals[:last+1]))
 	}
 	sort.Slice(data.Calls, func(i, j int) bool {
 		return data.Calls[i].Name < data.Calls[j].Name
 	})
 	return data
 }
 
+func newSyscallData(target *targets.Target, sc *prog.Syscall, attrs []uint64) SyscallData {
+	/*
+	 * The s390x Linux syscall ABI allows for upto 5 input parameters passed in registers, and this is not enough
+	 * for the mmap syscall. Therefore, all input parameters for the mmap syscall are packed into a struct
+	 * on user stack and the pointer to the struct is passed as an input parameter to the syscall. To temporarily
+	 * work around this problem we therefore cast the mmap syscall to the glibc mmap wrapper. This requires a proper
+	 * solution later.
+	 */
+	syscallTrampolines := map[string]map[string]map[uint64]string{
+		"linux": {
+			"s390x": {
```

Comment:
I think this will fit nicely as targets.Target member. Then this map can be defined just for linux/s390 and the code below can use just `target.SyscallTrampolines[sc.NR]`.
And I think we will need this for pkg/csource anyway, so this will also nicely avoid duplication.
But I think it's better to use call name as the key. 90 is not too readable/maintainable.

Reply:
done